### PR TITLE
Release v1.1.15

### PR DESCRIPTION
**Pull request by rata**
### Fixed

 * The `-ENOSYS` seccomp stub is now always generated for the native
   architecture that `runc` is running on. This is needed to work around some
   arguably specification-incompliant behaviour from Docker on architectures
   such as ppc64le, where the allowed architecture list is set to `null`. This
   ensures that we always generate at least one `-ENOSYS` stub for the native
   architecture even with these weird configs. (#4391)
 * On a system with older kernel, reading `/proc/self/mountinfo` may skip some
   entries, as a consequence runc may not properly set mount propagation,
   causing container mounts leak onto the host mount namespace. (#2404, #4425)

### Removed

 * In order to fix performance issues in the "lightweight" bindfd protection
   against [CVE-2019-5736], the temporary `ro` bind-mount of `/proc/self/exe`
   has been removed. runc now creates a binary copy in all cases. (#4392, #2532)

[CVE-2019-5736]: https://www.openwall.com/lists/oss-security/2019/02/11/2

---
I'd like to create a 1.1.15 release including the change to remove the bindfd logic (already backported :)), as that is causing us quite some pain. I'm of course fine if we release earlier than that date, in fact that would be great :)

I'll be on PTO starting today and coming back next week. Feel free to either push to this branch to amend any changes (maintainers can push to that branch on my behalf IIUC), otherwise I can address the changes early next week.

### CODE REVIEW

**Comment by rata** [comment]
Hmm, alma linux 9 is failing with:

```bash
not ok 14 runc run (cgroup v2 resources.unified only)
# (in test file tests/integration/cgroups.bats, line 270)
#   `[ "$status" -eq 0 ]' failed
# runc spec (status=0):
#
# runc run -d --console-socket /tmp/bats-run-eiy40K/runc.G1EDSZ/tty/sock test_cgroups_unified (status=1):
# time="2024-10-02T11:30:45Z" level=error msg="runc run failed: unable to start container process: container init was OOM-killed (memory limit too low?)"
```

**Comment by rata** [comment]
Fixed here for 1.1: https://github.com/opencontainers/runc/pull/4423

**Comment by kolyshkin** [comment]
Let's include #4425 into 1.1.15, too.

Also, we need to decide what to do with https://github.com/opencontainers/runc/issues/4347

**Comment by cyphar** [comment]
Since #4347 is a longer-standing issue I don't think we need to block a 1.1.z patch release on it. Maybe we might want to block 1.2.0 on it (depending on how the spec stuff goes) but I'm not sure if we would even want to backport a spec fix to 1.1.z.

**Comment by AkihiroSuda** [comment]
LGTM but needs rebase

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,27 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+   architecture that `runc` is running on. This is needed to work around some
+   arguably specification-incompliant behaviour from Docker on architectures
+   such as ppc64le, where the allowed architecture list is set to `null`. This
+   ensures that we always generate at least one `-ENOSYS` stub for the native
+   architecture even with these weird configs. (#4391)
+
+### Removed
+
+ * In order to fix performance issues in the "lightweight" bindfd protection
+   against [CVE-2019-5736][cve-2019-5736], the temporary `ro` bind-mount of
```
Comment (kolyshkin) [inline CHANGELOG.md:26]: The cve link is missing; I will add it.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,12 +6,31 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased 1.1.z]
 
+## [1.1.15] - 2024-10-07
```
Comment (kolyshkin) [inline CHANGELOG.md:9]: An anchor for this is missing (will add).

**Review by kolyshkin — APPROVED** [review]
LGTM

**Comment by kolyshkin** [comment]
I rebased this PR and did some minor fixes to the changelog (added missing CVE and 1.1.15 links, fixed the "Unreleased 1.1.z" link).

Also changed the release date to Monday, because why not.

**Comment by lifubang** [comment]
I searched PR with the label `kind/bug`, and found that #3546 is not only an new feature implementation, but also a bug in practise.
Do you think #3546 should be included?

**Review by AkihiroSuda — APPROVED** [review]

**Comment by kolyshkin** [comment]
> I searched PR with the label `kind/bug`, and found that #3546 is not only an new feature implementation, but also a bug in practise. Do you think #3546 should be included?

It's a relatively big one and might not be backportable as-is, so I'd rather have it in 1.2.0. If anything, this gives us yet another reason to release it sooner.

**Comment by rata** [comment]
Thanks!

**Pull request merged**